src/components/v5/frame/ColonyHome/partials/ReputationChart/hooks/useContributorsByDomain.tsx

```diff
@@ -0,0 +1,56 @@
+import { useMemo } from 'react';
+
+import { useMemberContext } from '~context/MemberContext/MemberContext.ts';
+import { type ColonyContributorFragment } from '~gql';
+import useGetSelectedDomainFilter from '~hooks/useGetSelectedDomainFilter.tsx';
+
+import { type ContributorItem } from '../types.ts';
+
+export const getContibutorsByDomain = (
+  members: ColonyContributorFragment[],
+  selectedTeamId: number | undefined,
+): ContributorItem[] => {
+  const isAllTeamsSelected = selectedTeamId === undefined;
+
+  return members.map((contributor) => {
+    const { contributorAddress, colonyReputationPercentage, user, reputation } =
+      contributor;
+
+    const teamReputationPercentage = reputation?.items?.find(
+      (item) => item?.domain?.nativeId === selectedTeamId,
+    )?.reputationPercentage;
+
+    return {
+      user,
+      walletAddress: contributorAddress,
+      reputation: isAllTeamsSelected
+        ? colonyReputationPercentage
+        : teamReputationPercentage,
+    };
+  });
+};
+
+export const useContributorsByDomain = () => {
+  const { totalMembers, loading } = useMemberContext();
+
+  const allContributors = useMemo(
+    () =>
+      totalMembers.filter(({ hasReputation }) => hasReputation).sort() ?? [],
+    [totalMembers],
+  );
+
+  const selectedDomain = useGetSelectedDomainFilter();
+
+  const contributorsWithReputationByDomain = useMemo(
+    () => getContibutorsByDomain(allContributors, selectedDomain?.nativeId),
+    [allContributors, selectedDomain],
+  );
+
+  const contributorsList = contributorsWithReputationByDomain.sort(
+    (a, b) => Number(b.reputation) - Number(a.reputation),
+  );
```

Comment:
Based on my testing, this list can contain contributors having `undefined` reputation, so we should refactor this

```suggestion
  const contributorsList = contributorsWithReputationByDomain
    .filter((contributor) => !!contributor.reputation)
    .sort((a, b) => Number(b.reputation) - Number(a.reputation));
```

Reply:
nice catch! It was there before refactoring, but I had wrong assumption about hasReputation filter. fixed :)